sotodlib/site_pipeline/make_atomic_filterbin_map.py

```diff
@@ -419,49 +378,45 @@ def main(config_file: str) -> None:
                                        subdir='temp_proc', remove=False)
 
     run_list = []
-    for oi in range(len(my_tods)):
-        # tod_list[oi].obslist[0] is the old obslist
-        pid = tod_list[oi].obslist[0][3]
-        detset = tod_list[oi].obslist[0][1]
-        band = tod_list[oi].obslist[0][2]
-        obslist = tod_list[oi].obslist
+    for oi, ol in enumerate(obslists_arr):
+        pid = ol[0][3]
+        detset = ol[0][1]
+        band = ol[0][2]
+        obslist = ol
         t = putils.floor(periods[pid, 0])
         t5 = ("%05d" % t)[:5]
         prefix = "%s/%s/atomic_%010d_%s_%s" % (
             args.odir, t5, t, detset, band)
-        if args.area is not None:
-            subshape, subwcs = subgeoms[oi]
 
         tag = "%5d/%d" % (oi+1, len(obskeys))
         putils.mkdir(os.path.dirname(prefix))
-        pwv_atomic = tod_list[oi].pwvs[0]
+        pwv_atomic = get_pwv(periods[pid, 0], periods[pid, 1], args.hk_data_path)
+
         # Save file for data base of atomic maps.
         # We will write an individual file,
         # another script will loop over those files
         # and write into sqlite data base
         if not args.only_hits:
-            info = []
+            info_list = []
             for split_label in split_labels:
-                info.append(bunch.Bunch(
-                    pid=pid,
-                    obs_id=obslist[0][0].encode(),
-                    telescope=obs_infos[obslist[0][3]].telescope.encode(),
-                    freq_channel=band.encode(),
-                    wafer=detset.encode(),
-                    ctime=int(t),
-                    split_label=split_label.encode(),
-                    split_detail=''.encode(),
-                    prefix_path=str(cwd + '/' + prefix + '_%s' %
-                                    split_label).encode(),
-                    elevation=obs_infos[obslist[0][3]].el_center,
-                    azimuth=obs_infos[obslist[0][3]].az_center,
-                    pwv=float(pwv_atomic)))
+                info = mapmaking.AtomicInfo(
+                    obslist[0][0],
+                    obs_infos[obslist[0][3]].telescope,
+                    band,
+                    detset,
+                    int(t))
```

Comment:
I would recommend using kwargs here instead of positional

Reply:
Could you elaborate a bit? I do not understand what you mean. Thanks

Reply:
I would have written:
```
info = mapmaking.AtomicInfo(
    obs_id=obslist[0][0],
    telescope=obs_infos[obslist[0][3]].telescope,
    ...
)
```
This allows you to change the order of parameters, etc, later without impacting this code in potentially invisible ways.